### PR TITLE
Added Modular Exponentiation [JAVA]

### LINKED ISSUE #135: Modular Exponentiation [JAVA]
[http://www.geeksforgeeks.org/modular-exponentiation-power-in-modular-arithmetic/](http://www.geeksforgeeks.org/modular-exponentiation-power-in-modular-arithmetic/)

### PR DESCRIPTION
**Pull request by aashutoshrathi**
Fixes #135 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous timespan of 7 days.

### CODE REVIEW

File: modular_exponential/ModularExponential.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class ModularExponential {
+	static int modularExponential(int base, int power, int mod) {
```
Comment (aashutoshrathi) [inline modular_exponential/ModularExponential.java]: @aviaryan should this issue be fixed

Comment (aviaryan) [inline modular_exponential/ModularExponential.java]: Yes, add public or private, whatever is correct.

Comment (aashutoshrathi) [inline modular_exponential/ModularExponential.java]: take a look now @aviaryan

File: modular_exponential/ModularExponential.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+public class ModularExponential {
+	public static int modularExponential(int base, int power, int mod) {
+		int result = 1;
```
Comment (aviaryan) [inline modular_exponential/ModularExponential.java]: Better to use BigInt here.

Comment (aashutoshrathi) [inline modular_exponential/ModularExponential.java]: just for `result` or for everything

Comment (aviaryan) [inline modular_exponential/ModularExponential.java]: Only for result should do.
But use a larger data type (long?) for the parameters.

**Comment by aashutoshrathi** [comment]
@aviaryan added BigInteger take a look

**Pull request merged**